server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -267,4 +277,128 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
             }
         });
     }
+
+    void setWeightForDecommissionedZone(List<String> zones, ActionListener<ClusterPutWRRWeightsResponse> listener) {
+        ClusterState clusterState = clusterService.getClusterApplierService().state();
+
+        DecommissionAttributeMetadata decommissionAttributeMetadata = clusterState.metadata().custom(DecommissionAttributeMetadata.TYPE);
+        assert decommissionAttributeMetadata.status().equals(DecommissionStatus.WEIGH_AWAY)
+            : "unexpected status encountered while decommissioning nodes";
+        DecommissionAttribute decommissionAttribute = decommissionAttributeMetadata.decommissionAttribute();
+
+        Map<String, String> weights = new HashMap<>();
+        zones.forEach(zone -> {
+            if (zone.equalsIgnoreCase(decommissionAttribute.attributeValue())) {
+                weights.put(zone, "0");
```

Comment:
is the weight string or float?

Reply:
Its float. Taken the latest Changes from WRR API PR. Done.